src/main.rs

```diff
@@ -465,6 +486,37 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_executable_on_command_line() {
+        let expect_env = call_env();
+        let d = get_current_dir().ok();
+        d.and_then(|p| {
+            let new_env = process::Command::new("randomtemp")
+                .env_remove("RANDOMTEMP_EXECUTABLE")
+                .env_remove("RANDOMTEMP_BASEDIR")
+                .env_remove("RANDOMTEMP_MAXTRIAL")
+                .arg(ENV_COMMAND)
+                .current_dir(p)
+                .output()
+                .expect("failed to execute process");
+
+            let new_output = new_env.stdout;
+            let new_error = new_env.stderr;
+
+            let expect_output = expect_env.stdout;
+            let expect_error = expect_env.stderr;
+
+            assert_ne!(expect_output, new_output);
+            assert_ne!(expect_output.len(), 0);
+            assert_ne!(new_output.len(), 0);
```

Comment:
I think it is somehow caused by the fact that the mechanism we used to run executables on Windows. Here `ENV_COMMAND` is recognized as a subprogram, so the argument is actually empty. When you run commands like `cmd /q /c %VAR%` where `VAR` is empty, it won't work.